packages/store/test/test-encodeKey.js

```diff
@@ -114,3 +124,30 @@ test('order invariants', t => {
     }
   }
 });
+
+test('BigInt values round-trip', async t => {
+  await fc.assert(
+    fc.property(fc.bigInt(), n => {
+      const rt = decodeKey(encodeKey(n));
+      return assertionPassed(t.is(rt, n), () => rt === n);
+    }),
+  );
+});
+
+test('BigInt encoding comparison corresponds with numeric comparison', async t => {
+  await fc.assert(
+    fc.property(fc.bigInt(), fc.bigInt(), (a, b) => {
+      if (a === b) return assertionPassed(t.pass(), () => true);
+      const ea = encodeKey(a);
+      const eb = encodeKey(b);
+      if (a < b) {
+        return assertionPassed(t.true(ea < eb), () => ea < eb);
+      }
+      if (a > b) {
+        return assertionPassed(t.true(ea > eb), () => ea > eb);
```

Comment:
Reproducing logic like this is super kludgy, but ava assertions returning a success/failure boolean was only added in v4 (cf. https://github.com/avajs/ava/releases/tag/v4.0.0#:~:text=Assertions%20as%20type%20guards ) and the upgrade appears to be non-trivial.